src/pages/ContentScripts/features/repo-activity-racing-bar/RacingBar.tsx

```diff
@@ -0,0 +1,201 @@
+import type { EChartsOption, EChartsType } from 'echarts';
+
+interface RacingBarProps {
+  repoName: string;
+  data: any;
+}
+
+// TODO generate color from user avatar
+const colorMap = new Map();
+
+const updateFrequency = 3000;
+
+const option: EChartsOption = {
+  grid: {
+    top: 10,
+    bottom: 30,
+    left: 150,
+    right: 50,
+  },
+  xAxis: {
+    max: 'dataMax',
+  },
+  yAxis: {
+    type: 'category',
+    inverse: true,
+    max: 10,
+    axisLabel: {
+      show: true,
+      fontSize: 14,
+      formatter: function (value: string) {
+        if (!value || value.endsWith('[bot]')) return value;
+        return `${value} {avatar${value.replaceAll('-', '')}|}`;
+      },
+    },
+    axisTick: {
+      show: false,
+    },
+    animationDuration: 0,
+    animationDurationUpdate: 200,
+  },
+  series: [
+    {
+      realtimeSort: true,
+      seriesLayoutBy: 'column',
+      type: 'bar',
+      itemStyle: {
+        color: function (params: any) {
+          const githubId = params.value[0];
+          if (colorMap.has(githubId)) {
+            return colorMap.get(githubId);
+          } else {
+            const randomColor =
+              '#' + Math.floor(Math.random() * 16777215).toString(16);
+            colorMap.set(githubId, randomColor);
+            return randomColor;
+          }
+        },
+      },
+      data: undefined,
+      encode: {
+        x: 1,
+        y: 0,
+      },
+      label: {
+        show: true,
+        precision: 1,
+        position: 'right',
+        valueAnimation: true,
+        fontFamily: 'monospace',
+      },
+    },
+  ],
+  // Disable init animation.
+  animationDuration: 0,
+  animationDurationUpdate: updateFrequency,
+  animationEasing: 'linear',
+  animationEasingUpdate: 'linear',
+  graphic: {
+    elements: [
+      {
+        type: 'text',
+        right: 60,
+        bottom: 60,
+        style: {
+          text: undefined,
+          font: 'bolder 60px monospace',
+          fill: 'rgba(100, 100, 100, 0.25)',
+        },
+        z: 100,
+      },
+    ],
+  },
+};
+
+const updateMonth = (instance: EChartsType, data: any, month: string) => {
+  const rich: any = {};
+  data[month].forEach((item: any[]) => {
+    // rich name cannot contain special characters such as '-'
+    rich[`avatar${item[0].replaceAll('-', '')}`] = {
+      backgroundColor: {
+        image: `https://avatars.githubusercontent.com/${item[0]}?s=48&v=4`,
+      },
+      height: 20,
+    };
+  });
+  // @ts-ignore
+  option.yAxis.axisLabel.rich = rich;
+  // @ts-ignore
+  option.series[0].data = data[month];
+  // @ts-ignore
+  option.graphic.elements[0].style.text = month;
+
+  // it seems that hidden bars are also rendered, so when each setOption merge more data into the chart,
+  // the fps goes down. So we use notMerge to avoid merging data. But this disables the xAxis animation.
+  // Hope we can find a better solution.
+  instance.setOption(option, {
+    notMerge: true,
+  });
+};
```

Comment:
microsoft/vscode has a lot of contributors, which slows down the animation when played at ~2018 or so.

![image](https://github.com/hypertrons/hypertrons-crx/assets/32434520/604c96c3-b2d8-4648-929c-06b6ee7bf1bb)

`notMerge: true` is the best resolution I can give for now.